### PR TITLE
Meta service framework

### PR DESCRIPTION
**Pull request by AmitRoushan**
The PR contains initial Metadata Service framework logic. 
- Metadata service is a gRPC server which accepts backup request from BackupController adn store on specified BackupLocation.
- It also receive request to fetch the backuped metadata and send it back to restore controller

Following are key highlights of the PR:
- [x] Initial framework of Meta service
- [x] Initial E2E test case
- [x] Client side GRPC load balancing logic

### CODE REVIEW

File: provider/meta_service/server/options/logging.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package options
```
Comment (sushanthakumar) [inline provider/meta_service/server/options/logging.go:1]: Suggest to move logging to common , can be to util directory

Comment (AmitRoushan) [reply provider/meta_service/server/options/logging.go]: @sushanthakumar logging options moved to ```utils```

File: provider/meta_service/Makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Copyright 2022 The SODA Authors.
```
Comment (sushanthakumar) [inline provider/meta_service/Makefile:1]: provider directory can be changed to providerframework for better clarity

**Comment by sushanthakumar** [comment]
Redundant PR, changes already taken care

**Pull request closed without merging**